metricshub-sql-extension/src/main/java/org/sentrysoftware/metricshub/extension/sql/SqlClientExecutor.java

```diff
@@ -0,0 +1,273 @@
+				log.error("Invalid name in lookup source table {}", sqlTable.getSource());
+				return null;
+			}
+
+			final String type = sqlColumn.getType();
+			if (type == null || type.isBlank()) {
+				log.error("Invalid type for column {} in lookup source table {}", name, sqlTable.getSource());
+				return null;
+			}
+
+			final int number = sqlColumn.getNumber();
+
+			// Check if number is valid
+			if (number < 1) {
+				log.error("Wrong number {} for column {} in lookup source table {}", number, name, sqlTable.getSource());
+				return null;
+			}
+			columnNameType.add(String.join(" ", name, type));
+		}
+
+		queryBuilder.append(String.join(",", columnNameType));
+		queryBuilder.append(");");
+
+		return queryBuilder.toString();
+	}
+
+	/**
+	 * Create the INSERT SQL query needed to insert the values in the {@link SqlTable} into a SQL Database.
+	 * @param sqlTable The table to insert the data from.
+	 * @return The resulting SQL query.
+	 */
+	private String insertTableQuery(final SqlTable sqlTable) {
+		StringBuilder queryBuilder = new StringBuilder();
+		queryBuilder.append("INSERT INTO ");
+
+		final String alias = sqlTable.getAlias().strip();
+
+		queryBuilder.append(alias);
+		queryBuilder.append(" (");
+
+		final List<String> columnNames = new ArrayList<>();
+
+		sqlTable.getColumns().stream().forEach(sqlColumn -> columnNames.add(sqlColumn.getName()));
+
+		queryBuilder.append(String.join(",", columnNames));
+		queryBuilder.append(") VALUES ");
+
+		final String valuesToInsert = formatInsertValues(sqlTable);
+
+		// If there is no value to insert, we stop the job completely
+		if (valuesToInsert == null) {
+			return null;
+		}
+
+		queryBuilder.append(valuesToInsert);
+		queryBuilder.append(";");
+
+		return queryBuilder.toString();
+	}
+
+	/**
+	 * Extract the {@link SourceTable} from a {@link SqlTable}, extract its data and format them for a INSERT SQL query.
+	 * @param sqlTable The table to extract the {@link SourceTable} and the {@link SqlColumn} from.
+	 * @return The formated data.
+	 */
+	private String formatInsertValues(final SqlTable sqlTable) {
+		SourceTable sourceTable = SourceTable.lookupSourceTable(sqlTable.getSource(), connectorId, telemetryManager).get();
+
+		if (sourceTable == null) {
+			log.error("The source table {} is not found during the SQL Query job. Skip processing.", sqlTable.getSource());
+			return null;
+		}
+
+		List<List<String>> table = sourceTable.getTable();
+
+		if (table == null || table.isEmpty()) {
+			log.error("The source table {} is empty. Skip SQL job processing.", sqlTable.getSource());
+			return null;
+		}
+
+		final List<SqlColumn> sqlColumns = sqlTable.getColumns();
+
+		final List<String> columnValues = new ArrayList<>();
+
+		for (final List<String> row : table) {
+			final List<String> rowValues = new ArrayList<>();
+			for (final SqlColumn sqlColumn : sqlColumns) {
+				final String separator = sqlColumn.getType().contains("VARCHAR") ? "'" : "";

```

Comment:
Getting  the value at the beginning is better `String value = row.get(sqlColumn.getNumber() - 1);`